esp-wifi/src/wifi/os_adapter.rs

```diff
@@ -33,42 +33,56 @@ use crate::compat::common::syslog;
 
 use super::WifiEvent;
 
-pub(crate) static mut WIFI_STATE: i32 = -1;
+pub(crate) static mut STA_STATE: WifiState = WifiState::Invalid;
```

Comment:
A bit off-topic from the PR, but do we need any synchronization here? I wonder if there are any race conditions around checking the state and the state and then setting the state. 

It might be possible to use an atomic variable here (which may end up undoing some of the work here in this PR) or a plain ol' mutex would work.

Reply:
We can do some atomic-enum shenanigans here for sure. It has never been an issue, though, or at least none that caused enough pain

Reply:
That being said, I'm not sure how well atomic-enum would work with the S2... worth a try, but I'm not opposed to unpretty this some if this point is a concern